### PR TITLE
Make goto async

### PR DESCRIPTION
**Pull request by u9g**

### CODE REVIEW

**Comment by u9g** [comment]
might want to hold off on this, as it's probably a better idea to make a new gotoasync() function instead of just rewriting goto() as that would break existing projects

**Review by rom1504 — CHANGES_REQUESTED** [review]
Can you use the same technique we used in mineflayer ?
Ie make it async but use a callbackify function on top so it both returns a promise and take a callback.

File: lib/goto.js [hunk 1]
```diff
@@ -1,47 +1,57 @@
-function error (name, message) {
-  const err = new Error(message)
-  err.name = name
-  return err
-}
+// function error (name, message) {
+//   const err = new Error(message)
+//   err.name = name
+//   return err
+// }
```
Comment (Karang) [inline lib/goto.js:5]: If this function is not needed, remove it, don't comment.

File: lib/goto.js [hunk 2]
```diff
@@ -1,47 +1,57 @@
...
 
-  function noPathListener (results) {
-    if (results.status !== 'noPath') return
-    cleanup(error('NoPath', 'No path to the goal!'))
-  }
+    function noPathListener (results) {
+      if (results.status !== 'noPath') {
+        // cleanup(error('NoPath', 'No path to the goal!'))
+      }
+    }
```
Comment (Karang) [inline lib/goto.js:26]: This callback does nothing. It should reject in case the status is noPath.

File: lib/goto.js [hunk 3]
```diff
@@ -1,47 +1,57 @@
...
 
-  function goalChangedListener (newGoal) {
-    if (newGoal === goal) return
-    cleanup(error('GoalChanged', 'The goal was changed before it could be completed!'))
-  }
+    function goalChangedListener (newGoal) {
+      if (newGoal === goal) {
+        // cleanup(error('GoalChanged', 'The goal was changed before it could be completed!'))
+      }
+    }
```
Comment (Karang) [inline lib/goto.js:32]: if the goal is changed newGoal !== goal, it should reject the promise

**Review by Karang — CHANGES_REQUESTED** [review]

**Pull request merged**